lib/gateway/server.js

```diff
@@ -20,9 +20,14 @@ function createTlsServer (httpsConfig, app) {
   let defaultCert = null;
   let sniCerts = [];
 
-  for (let el in httpsConfig.tls) {
-    let domain = el;
-    let certPaths = httpsConfig.tls[el];
+  let domainCount = httpsConfig.tls.length;
+  const domainNames = httpsConfig.tls
```

Comment:
@kevinswiber @larryschirmer 

this fails in Circle CI if not tls provided in https section. so additional check is needed. 

It seems there is a variant TLS-PSK that does not need a certificate. And it seems to be not supported by node.js https://github.com/nodejs/node/pull/6701

So until then certificates are required. And I would imaging throwing an error and preventing start would be a good option.